### PR TITLE
Mes 3972 category specific fault count provider methods

### PR DESCRIPTION
**Pull request by ammarhaiderbjss**
## Description

- Creates category specific methods for fault sum counting (in `src/providers/fault-count/fault-count.ts`)
- Exposes higher order functions that accept category as a parameter and call the correct category method
- Implements usage of the existing `TestCategory` enum across the app 

## Checklist

- [ ] PR title includes the JIRA ticket number
- [ ] Branch is rebased against the latest develop
- [ ] Code has been tested manually
- [ ] PR link added to JIRA ticket
- [ ] One review from each scrum team
- [ ] Squashed commit contains the JIRA ticket number

### CODE REVIEW

File: src/pages/debrief/cat-be/__tests__/debrief.cat-be.page.spec.ts [hunk 1]
```diff
@@ -71,6 +72,7 @@ describe('DebriefCatBEPage', () => {
             startedTests: {
               123: {
                 testSlotAttributes,
+                category: TestCategory.B,
```
Comment (rossfauldsbjss) [inline src/pages/debrief/cat-be/__tests__/debrief.cat-be.page.spec.ts]: should this be BE or is that functionality for the future?

**Review by rossfauldsbjss — APPROVED** [review]
Nice work!

File: src/providers/fault-count/fault-count.ts [hunk 2]
```diff
@@ -81,7 +102,65 @@ export class FaultCountProvider {
...
+  }
+
+  public getSeriousFaultSumCountCatBE = (data: CatBUniqueTypes.TestData): number => {
+
+    // The way how we store serious faults differs for certain competencies
+    // Because of this we need to pay extra attention on summing up all of them
+    const { seriousFaults, manoeuvres, vehicleChecks, eyesightTest } = data;
+
+    const seriousFaultSumOfSimpleCompetencies = Object.keys(pickBy(seriousFaults)).length;
+    const vehicleCheckSeriousFaults = vehicleChecks.showMeQuestion.outcome === CompetencyOutcome.S ? 1 : 0;
```
Comment (rstepnowski) [inline src/providers/fault-count/fault-count.ts:134]: I thought that for B+E tests vehicleChecks.showMeQuestion is actually an array - is that not the case anymore? 
How and when the Serious marker will be checked? Will it be reverted if DE will re-ask the questions again (by going back to that page and amending the results)

File: src/providers/fault-count/fault-count.ts [hunk 3]
```diff
@@ -81,7 +102,65 @@ export class FaultCountProvider {
...
+
+    const result =
+      drivingFaultSumOfSimpleCompetencies +
+      this.sumManoeuvreFaults(manoeuvres, CompetencyOutcome.DF) +
+      this.sumVehicleCheckFaultsCatBE(vehicleChecks);
+
+    return result;
+  }
+
+  public getSeriousFaultSumCountCatBE = (data: CatBUniqueTypes.TestData): number => {
```
Comment (rstepnowski) [inline src/providers/fault-count/fault-count.ts:127]: CatBUniqueTypes  -> Cat**BE**UniqueTypes

File: src/providers/fault-count/fault-count.ts [hunk 4]
```diff
@@ -81,7 +102,65 @@ export class FaultCountProvider {
...
+    const result =
+      seriousFaultSumOfSimpleCompetencies +
+      this.sumManoeuvreFaults(manoeuvres, CompetencyOutcome.S) +
+      vehicleCheckSeriousFaults +
+      eyesightTestSeriousFaults;
+
+    return result;
+  }
+
+  public getDangerousFaultSumCountCatBE = (data: CatBUniqueTypes.TestData): number => {
```
Comment (rstepnowski) [inline src/providers/fault-count/fault-count.ts:146]: as above -> type hinting  typo

File: src/providers/fault-count/fault-count.ts (hunk 3 above)
Comment (ammarhaiderbjss) [reply src/providers/fault-count/fault-count.ts:127]: Have another sub-task to update this when https://github.com/dvsa/mes-mobile-app/pull/920 goes in..

File: src/providers/fault-count/fault-count.ts (hunk 2 above)
Comment (ammarhaiderbjss) [reply src/providers/fault-count/fault-count.ts:134]: This will be updated when https://github.com/dvsa/mes-mobile-app/pull/920 goes in..

Comment (ammarhaiderbjss) [reply src/providers/fault-count/fault-count.ts:134]: That contains the logic to do this B+E specific calculation for Vehicle Checks, next PR will incorporate that logic in here as well.

File: src/providers/fault-count/__mocks__/catBETestDataStateObject.ts [hunk 5]
```diff
@@ -0,0 +1,53 @@
+import { QuestionResult } from '@dvsa/mes-test-schema/categories/BE/partial';
+import { CatBEUniqueTypes } from '@dvsa/mes-test-schema/categories/BE';
+
+export const catBETestDataStateObject: CatBEUniqueTypes.TestData = {
```
Comment (szabikr) [inline src/providers/fault-count/__mocks__/catBETestDataStateObject.ts]: We should use **kebab-case**-ing for the file name

File: src/providers/fault-count/__mocks__/catBTestDataStateObject.ts [hunk 6]
```diff
@@ -0,0 +1,51 @@
+import { CompetencyOutcome } from '../../../shared/models/competency-outcome';
+import { ManoeuvreOutcome } from '@dvsa/mes-test-schema/categories/Common';
+import { CatBUniqueTypes } from '@dvsa/mes-test-schema/categories/B';
+
+export const catBTestDataStateObject: CatBUniqueTypes.TestData = {
```
Comment (szabikr) [inline src/providers/fault-count/__mocks__/catBTestDataStateObject.ts]: **kebab-case**-ing for the file name

**Review by szabikr — COMMENTED** [review]
A few of the category B+E specific functions use CatBUniqueTypes still

**Review by szabikr — APPROVED** [review]

**Pull request merged**